### PR TITLE
Split up more & general additions

### PR DESCRIPTION
**Pull request by VandeurenGlenn**
- move coinmarketcap & github fetchers to api package
- add initial ws & http clients
- moves cache to it's own package
- simplify github repos fetch
- add stats etc to server repo (for now)

### CODE REVIEW

**Comment by vdegenne** [comment]
Yes I think the cache package should go back into the `api` package

**Comment by VandeurenGlenn** [comment]
> Yes I think the cache package should go back into the `api` package

Done!

**Comment by vdegenne** [comment]
> > Yes I think the cache package should go back into the `api` package
> 
> Done!
> 
> edit: o wait, api? I've putted it back in server

Nah I think it makes sense in `api` because that's where you are caching things from no?

**Comment by vdegenne** [comment]
(I wanted to do a test, I can push on your branch, that's what I wanted to know.)

**Review by vdegenne — COMMENTED** [review]
cool

**Review by vdegenne — APPROVED** [review]

**Comment by vdegenne** [comment]
Let's merge so we can start branching `main`

File: packages/api/src/client/ws/index.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+export class WSApiClient {
+  // @ts-ignore
+  #client;
+
+  constructor() {
+    this.#init() 
+  }
+
+  async #init() {
+    this.#client = new SocketRequestClient('ws://blue.leofcoin.org', 'protocol-blue')
```
Comment (vdegenne) [inline packages/api/src/client/ws/index.ts:15]: ERRORS!!!!!

Comment (vdegenne) [reply packages/api/src/client/ws/index.ts:15]: let's dismiss that for now

Comment (VandeurenGlenn) [reply packages/api/src/client/ws/index.ts:15]: Yeah, I need to fix declarations, the module itself works.

**Pull request merged**